src/API/NoteActions.php

```diff
@@ -115,6 +115,24 @@ public function trigger_note_action( $request ) {
 			$tracks_event = 'wcadmin_inbox_action_click';
 		}
 
+		$screen_name = '';
+
+		if ( isset( $_SERVER['HTTP_REFERER'] ) ) {
+			parse_str( wp_parse_url( $_SERVER['HTTP_REFERER'], PHP_URL_QUERY ), $queries ); // phpcs:ignore sanitization ok.
+		}
+		if ( isset( $queries ) ) {
+			$page      = isset( $queries['page'] ) ? $queries['page'] : null;
+			$path      = isset( $queries['path'] ) ? $queries['path'] : null;
+			$post_type = isset( $queries['post_type'] ) ? $queries['post_type'] : null;
```

Comment:
This works well for post type list page, but we won't be able to see the type from individual posts this way.  For example, clicking into an individual order or product and actioning a note, the `screen` property will be blank.

I'm not sure how important this is for tracking (feel free to weigh in @pmcpinto), but if we need to solve this, we could grab the post type from the ID.

```php
get_post_type( $queries['post'] );
```

Reply:
Nice catch @joshuatf! Thank you 🙂
I added the method you mentioned in the commit 2636420